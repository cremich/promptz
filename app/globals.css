pre,
code {
  word-wrap: break-word !important;
  white-space: pre-wrap !important;
  white-space: -moz-pre-wrap !important;
  white-space: -pre-wrap !important;
  white-space: -o-pre-wrap !important;
  margin: 0;
  padding: 0;
}

code {
  background-color: #f3f3f7;
}

pre > code {
  display: inline-block;
  padding: 1em;
}
