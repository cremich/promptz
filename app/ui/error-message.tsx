import { Alert, AlertTitle, AlertDescription } from "@/components/ui/alert";
import { Terminal } from "lucide-react";

export type ErrorMessageProps = {
  title?: string;
  description: string;
};

export function ErrorMessage(props: ErrorMessageProps) {
  return (
    <Alert className="bg-rose-500">
      <Terminal className="h-4 w-4" />
      <AlertTitle>Heads up! {props.title}</AlertTitle>
      <AlertDescription>{props.description}</AlertDescription>
    </Alert>
  );
}
