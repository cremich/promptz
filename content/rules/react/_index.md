---
title: "React"
description: "Rules for React development, component patterns, and best practices."
---
