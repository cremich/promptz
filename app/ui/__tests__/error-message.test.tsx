import { describe, expect, test } from "@jest/globals";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { ErrorMessage } from "@/app/ui/error-message";

describe("Error Message", () => {
  test("renders error message unchanged", () => {
    const { container } = render(<ErrorMessage description="error" />);
    expect(container).toMatchSnapshot();
  });
});
