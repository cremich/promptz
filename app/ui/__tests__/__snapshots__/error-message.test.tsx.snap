// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Message renders error message unchanged 1`] = `
<div>
  <div
    class="relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg]:text-foreground text-foreground bg-rose-500"
    role="alert"
  >
    <svg
      class="lucide lucide-terminal h-4 w-4"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="4 17 10 11 4 5"
      />
      <line
        x1="12"
        x2="20"
        y1="19"
        y2="19"
      />
    </svg>
    <h5
      class="mb-1 font-medium leading-none tracking-tight"
    >
      Heads up! 
    </h5>
    <div
      class="text-sm [&_p]:leading-relaxed"
    >
      error
    </div>
  </div>
</div>
`;
